### PR TITLE
Website: Document script and style versioning support.

### LINKED ISSUE #211: Document script/style loading behavior and benefits

### PR DESCRIPTION
**Pull request by nicksay**
- Describe unique URLs and hashing of inline resource text.
- Describe events fired when unloading.

Closes #211.

### CODE REVIEW

File: web/documentation/versioning.md [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+> quick hash of the text content.  Whitespace is ignored when
+> calculating the hash, so indentation and formatting changes
+> will not trigger updates.
+
+
+## Resource events
+
+If you need to handle resources being unloaded when a new
+version is detected, (e.g. to dispose event listeners, etc), SPF
+will dispatch events before and after a resources is removed.
```
Comment (awbraunstein) [inline web/documentation/versioning.md]: s/resources/resource/g

File: web/documentation/versioning.md [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+As with those detailed in the [Events][events] documentation,
+these events are defined in the [API][] as [spf.Event][]
+objects.  The events and their descriptions follow:
+
+**`spfcssbeforeunload`**  
+Fired before unloading an managed style resource.  Occurs when
+an updated style is discovered for a given name.
+
+**`spfcssunload`**  
+Fired when unloading an managed style resource.  If the style
```
Comment (awbraunstein) [inline web/documentation/versioning.md]: unloading a managed

File: web/documentation/versioning.md [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+is being unloaded as part of switching versions, unloading of
+the old style occurs after the new style is loaded to avoid
+flashes of unstyled content.
+
+**`spfjssbeforeunload`**  
+Fired before unloading an managed script resource.  Occurs when
+an updated script is discovered for a given name.
+
+**`spfjsunload`**  
+Fired when unloading an managed script resource.  If the script
```
Comment (awbraunstein) [inline web/documentation/versioning.md]: unloading a managed

File: web/documentation/versioning.md (hunk 1 above)
Comment (nicksay) [inline web/documentation/versioning.md]: Done

File: web/documentation/versioning.md (hunk 2 above)
Comment (nicksay) [inline web/documentation/versioning.md]: Done

File: web/documentation/versioning.md (hunk 3 above)
Comment (nicksay) [inline web/documentation/versioning.md]: Done

**Comment by nicksay** [comment]
PTAL

**Comment by awbraunstein** [comment]
lgtm

**Pull request merged**